lib/web_ui/test/text_editing_test.dart

```diff
@@ -731,19 +734,28 @@ void testMain() {
       // DOM element is blurred.
       textEditing.editingElement.domElement.blur();
 
-      expect(spy.messages, hasLength(0));
-
-      // DOM element still has focus.
-      // Even though this passes on manual tests it does not work on
-      // Firefox automated unit tests.
-      if (browserEngine != BrowserEngine.firefox) {
-        expect(document.activeElement, textEditing.editingElement.domElement);
+      // For ios-safari the connection is closed.
+      if (browserEngine == BrowserEngine.webkit &&
+          operatingSystem == OperatingSystem.iOs) {
+        expect(spy.messages, hasLength(1));
+        expect(spy.messages[0].channel, 'flutter/textinput');
+        expect(spy.messages[0].methodName,
+            'TextInputClient.onConnectionClosed');
+        // DOM element loses the focus.
+        Timer.run(expectAsync0(() {
+          expect(document.activeElement, document.body);
+        }));
```

Comment:
I think there's a better way to do this with fake async. @yjbanov ?

Reply:
I read the dart documentation, this was the preferred method.

@yjbanov any ideas, this is the page that I followed: https://api.flutter.dev/flutter/test_api/expectAsync.html that directed me to the `expectAsync0`.

Reply:
If the test relies on real async DOM operations, those cannot be faked, so you might need real timers. I'm wondering, however, if in this case this will work too:

```dart
await Future<void>.delayed(Duration.zero);
expect(...);
```

I think this pattern is more common, but I'm not sure if it works in this test.

Reply:
Thanks for the comments!

Both works. I believe the reason that `expectAsync0` is less popular might be due to it is relative young age :) 

I changed the code to `await Future<void>.delayed(Duration.zero);` to keep in sync with other web_ui code.